### DISCUSSION TITLE
Possible to hide APIDemoPanel?

### DISCUSSION THREAD

**Discussion by IanVS** [PaloAltoNetworks/docusaurus-openapi-docs#414] [Q&A]
I'd like to remove the APIDemoPanel for now, what's the best way to do that?  I don't see an option in the configuration settings, so I guess it's down to custom CSS to hide it, but it would be nice not to generate it at all, if possible.

**Comment by sserrata** [comment]
There is one way. Basically, you'd have to override the `template` used to generate the `.api.mdx` files, using [this](https://docusaurus-openapi.tryingpan.dev/#config:~:text=requests%20from%20browser.-,template,-string) option.

Here's a full template:

https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/blob/main/demo/api.mustache


```
---
id: {{{id}}}
title: "{{{title}}}"
description: "{{{frontMatter.description}}}"
{{^api}}
sidebar_label: Introduction
{{/api}}
{{#api}}
sidebar_label: "{{{title}}}"
{{/api}}
{{^api}}
sidebar_position: 0
{{/api}}
hide_title: true
{{#api}}
hide_table_of_contents: true
{{/api}}
{{#json}}
api: {{{json}}}   // omitting this should hide the demo panel, since it renders only when `frontMatter.api` is present
{{/json}}
{{#api.method}}
sidebar_class_name: "{{{api.method}}} api-method"
{{/api.method}}
{{#infoPath}}
info_path: {{{infoPath}}}
{{/infoPath}}
custom_edit_url: null
{{#frontMatter.proxy}}
proxy: {{{frontMatter.proxy}}}
{{/frontMatter.proxy}}
---

{{{markdown}}}
```

**Reply by IanVS** [reply]
Unfortunately, that doesn't seem to work.  If I remove the line with your comment, I get:

```
This page crashed.
could not find react-redux context value; please ensure the component is wrapped in a <Provider>
```

I also tried removing the lines above and below that one, with the same result.

**Reply by sserrata** [reply]
You're correct - this used to work before when redux was only used in the API Demo Panel. It breaks because `MimeTabs` require the redux `Provider`. That said, operations with no mime/media tabs end up defaulting to `DocItem` instead of `ApiItem` due to the logic that distinguishes API vs non-API docs, which is also probably not what you want.

Example:

<img width="1437" alt="Screenshot 2023-02-16 at 9 23 53 AM" src="https://user-images.githubusercontent.com/9343811/219391092-8737ddb4-7257-4d0c-9eac-fc8c167603e3.png">

**Reply by IanVS** [reply]
Alright, I'll just stick with good ol' `display: none;` for now then.
